docs/nddata/covariance.rst

```diff
@@ -0,0 +1,187 @@
+
+Covariance
+**********
+
+Overview
+========
+
+The :class:`~astropy.nddata.covariance.Covariance` object is a general utility
+for constructing, visualizing, and storing two-dimensional covariance matrices.
+
+.. _nddata-covariance-construction:
+
+Construction
+============
+
+Beyond the nominal instantiation method, there are numerous methods used to
+construct a :class:`~astropy.nddata.covariance.Covariance` object.
+
+ #. The simplest approaches are if you have a variance vector or a
+    pre-calculated covariance matrix as a dense array and you want to construct
+    a :class:`~astropy.nddata.covariance.Covariance` object for further
+    calculations::
+
+    >>> import numpy as np
+    >>> from astropy.nddata.covariance import Covariance
+    >>> var = np.ones(3, dtype=float)
+    >>> covar = Covariance.from_variance(var)
+    >>> np.array_equal(covar.toarray(), np.identity(3))
+    True
+    >>> c = (np.diag(np.full(10-2, 0.2, dtype=float), k=-2)
+    ...         + np.diag(np.full(10-1, 0.5, dtype=float), k=-1)
+    ...         + np.diag(np.full(10, 1.0, dtype=float), k=0)
+    ...         + np.diag(np.full(10-1, 0.5, dtype=float), k=1)
+    ...         + np.diag(np.full(10-2, 0.2, dtype=float), k=2))
+    >>> covar = Covariance.from_array(c)
+    >>> np.array_equal(covar.toarray(), c)
+    True
+    >>> covar.toarray()
+    array([[1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. , 0. , 0. ],
+           [0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. , 0. ],
+           [0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. ],
+           [0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. ],
+           [0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. ],
+           [0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. ],
+           [0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. ],
+           [0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2],
+           [0. , 0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5],
+           [0. , 0. , 0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. ]])
+
+    Note the use of the :func:`~astropy.nddata.covariance.Covariance.toarray`
+    above to access the array; see :ref:`nddata-covariance-access`.
+
+ #. You can construct a covariance matrix based on samples from a
+    distribution::
+
+    >>> import numpy as np
+    >>> from mangadap.util.covariance import Covariance
```

Comment:
Is `mangadap` a new package? I don't think I want that as doc build dependency in astropy. Are you able to build an example without this specialized package?

Reply:
Sorry!  I missed editing this to import from astropy instead.  Fixed.